src/sgame/sg_weapon.cpp

```diff
@@ -142,9 +142,9 @@ static bool CanUseAmmoRefill( gentity_t *self )
 		// clipless weapons can be refilled whenever they lack ammo
 		return ( ps->ammo != wa->maxAmmo );
 	}
-	else if ( ps->clips != wa->maxClips )
+	else if ( ps->clips != wa->maxClips || ps->ammo != wa->maxAmmo )
```

Comment:
There is no longer a need to special-case clipless weapons. This if/else if/else chain can be replaced with `return ps->clips != wa->maxClips || ps->ammo != wa->maxAmmo`